core/prisma/migrations/20250305135055_add_action_ref_to_rul/migration.sql

```diff
@@ -0,0 +1,13 @@
+-- AlterTable
+ALTER TABLE "rules" ADD COLUMN     "watchedActionId" TEXT;
+
+-- AddForeignKey
+ALTER TABLE "rules" ADD CONSTRAINT "rules_watchedActionId_fkey" FOREIGN KEY ("watchedActionId") REFERENCES "action_instances"("id") ON DELETE SET NULL ON UPDATE CASCADE;
+
+DROP INDEX "rules_actionInstanceId_event_key";
+
+-- unique index for action chaining events
+CREATE UNIQUE INDEX "unique_action_chaining_events" ON "rules" ("actionInstanceId", "event", "watchedActionId") WHERE "watchedActionId" IS NOT NULL;
+
+-- unique index for regular events
+CREATE UNIQUE INDEX "unique_regular_events" ON "rules" ("actionInstanceId", "event") WHERE "watchedActionId" IS NULL;
```

Comment:
instead of having a unique index for rules of `actionInstanceId` + `event`, now there's one for `actionInstanceId + event + watchedActionInstanceId = null` and one for `actionInstanceId + event + watchedActionInstanceId`.

Maybe i could have made it more specific by only allowing  the latter index with `watchedActionInstanceId` for `event.actionSucceeded` and `event.actionFailed`. let me know if i should change that